### PR TITLE
feat(stories): new function to generate story template args

### PR DESCRIPTION
**Pull request by Supamiu**
## Description

The new `generateInputs` global function allows to create template inputs based on the story's context. This allows for cleaner template examples that won't include inputs if their assigned value is the defautl value for instance.

-----

The PR is full of junk because it was created before the rc commit issues happened, I can squash them all in one commit if needed but that's going to be done when merging so yeah, better just check files changed category.

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2428/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/10089/ui-diff)

File: stories/helpers/stories.ts [hunk 1]
```diff
@@ -84,3 +85,36 @@ export function cleanupTemplate(template: string): string {
...
+		if (value === defaultValue || value === null || value === undefined) {
+			return acc;
+		}
+		// Let's treat boolean inputs as booleanAttributes for stories
+		if (!disableBooleanAttributes && typeof value === 'boolean') {
+			if (value) {
+				return `${acc} ${name}`;
+			}
+			return acc;
+		}
```
Comment (GuillaumeNury) [inline stories/helpers/stories.ts:107]: Can you explain this choice? Do we have any way of knowing if it is a transformed input or not?

File: stories/helpers/stories.ts [hunk 2]
```diff
@@ -84,3 +85,36 @@ export function cleanupTemplate(template: string): string {
...
+		return `${acc} ${name}="${value.toString()}"`;
+	}, '');
+}
+
+export function storyInput<T>(inputName: string, currentValue: T, defaultValue: T, booleanAttribute = false): string {
```
Comment (GuillaumeNury) [inline stories/helpers/stories.ts:112]: Unused?

**Review by GuillaumeNury — COMMENTED** [review]
Nice! Just two questions :)

File: stories/helpers/stories.ts (hunk 1 above)
Comment (Supamiu) [reply stories/helpers/stories.ts:107]: What do you mean by transformed input in this context?

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/11326/ui-diff)

File: stories/helpers/stories.ts (hunk 1 above)
Comment (GuillaumeNury) [reply stories/helpers/stories.ts:107]: Something like `@Input({ transform: someTransformer })`

**Review by jeremie-lucca — APPROVED** [review]

**Pull request merged**